src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs

```diff
@@ -4875,6 +4875,57 @@ public Goo(StackOverflowException e)
     expectedContainers: ImmutableArray<string>.Empty,
     expectedDocumentName: "Goo.cs");
         }
+
+        [WorkItem(22293, "https://github.com/dotnet/roslyn/issues/22293")]
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        [InlineData("void")]
+        [InlineData("int")]
+        public async Task TestMethodGroupWithMissingSystemActionAndFunc(string returnType)
+        {
+            await TestInRegularAndScriptAsync(
+    $@"
+<Workspace>
+    <Project Language=""C#"" CommonReferences=""false"">
+        <Document><![CDATA[
+class C
+{{
+    void M()
+    {{
+        new [|Class|](Method);
+    }}
+
+    {returnType} Method()
+    {{
+    }}
+}}
+]]>
+        </Document>
+    </Project>
+</Workspace>",
+    $@"
+class C
+{{
+    void M()
+    {{
+        new Class(Method);
+    }}
+
+    {returnType} Method()
+    {{
+    }}
+}}
+
+internal class Class
+{{
+    private global::System.Object method;
```

Comment:
this must be caused by CommonReferences="false" for some reason... in VS with .NET 3.0 (as specified in the issue) it just uses `object`

Reply:
Yup.